app/components/reporting/reporting-roles-edit/reporting-roles-edit-directive.js

```diff
@@ -58,9 +62,40 @@ angular.module('roles.edit.directive', [
           return false;
         };
 
+        $scope.getDirtyRates = function () {
+          var dirty = [];
+          // // Validate if flat_rate is dirty
+          if ($scope.model.hasOwnProperty('flat_rate') && !_.isEmpty($scope.model.flat_rate.$dirty())) {
+            dirty.push($scope.model.flat_rate);
+          }
+
+          // Validate if hourly_rate is dirty
+          if ($scope.model.hasOwnProperty('hourly_rate')) {
+            if (!_.isEmpty($scope.model.hourly_rate.$dirty())) {
```

Comment:
I originally had it as one statement, but the hourly_rate property is set post fetch attempt, so it results in an error